docs/proposals/dedicated-nodes.md

```diff
@@ -0,0 +1,210 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/dedicated-nodes.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Dedicated Nodes, Taints, and Tolerations
+
+## Introduction
+
+This document proposes an approach to implementing *dedicated nodes* in Kubernetes.
+A *dedicated node* is a node that can only be used by a limited set of users.
+Dedicated nodes usually occur in sets, with each set of dedicated nodes being
+associated with a particular set of users. There are many possible generalizations of dedicated
+nodes, for example a set of nodes with a particular piece of hardware could be dedicated to
+pods that require that hardware.
+
+In theory, dedicated nodes can be implemented using
+[NodeConstraint](https://github.com/kubernetes/kubernetes/pull/18261). But `NodeConstraint`
+is not well-suited for this use because `NodeConstraint` cannot easily express the idea that
+*users not in the authorized set are excluded from the dedicated nodes*, which is
+a key property of the dedicated nodes concept. Expressing
+such an exclusion rule using `NodeConstraint` would work something like the following.
+First you would label every dedicated node with `{key: "dedicated", value: "foo"}`
+where `foo` is the name of the dedicated group to whom the node is dedicated. This is
+easy. But then you would have to add to the PodSpec of every pod created by every user
+a `NodeConstraint` like `"dedicated" NotIn A,B,C,D,...` where
```

Comment:
If all nodes have a dedicated label, then all pods can have a simple equality node constraint.

For example, suppose we have dedicated machine set "gpus".
Label those machines "dedicated=gpus", and label the rest of the machines "dedicated=public".
Then all pods just need a `NodeConstraint: dedicated=gpus` or `NodeConstraint: dedicated=public`.